ownCloud/Import/ImportFilesController.swift

```diff
@@ -0,0 +1,154 @@
+
+		if bookmarks.count > 1 {
+			let moreViewController = self.cardViewController(for: url)
+			if let wd = UIApplication.shared.delegate?.window {
+				let vc = wd!.rootViewController
+				if let navCon = vc as? UINavigationController, let topVC = navCon.visibleViewController {
+					OnMainThread {
+						topVC.present(asCard: moreViewController, animated: true)
+					}
+				} else {
+					OnMainThread {
+						vc?.present(asCard: moreViewController, animated: true)
+					}
+				}
+
+				return true
+			}
+		} else if bookmarks.count == 1, let bookmark = bookmarks.first {
+			self.importItemWithDirectoryPicker(with: url, into: bookmark)
+
+			return true
+		}
+
+		return false
+	}
+
+	func importItemWithDirectoryPicker(with url : URL, into bookmark: OCBookmark) {
+		OCCoreManager.shared.requestCore(for: bookmark, setup: { (_, _) in
+		}, completionHandler: { (core, error) in
+			if error == nil {
+				OnMainThread {
+					let directoryPickerViewController = ClientDirectoryPickerViewController(core: core!, path: "/", selectButtonTitle: "Save here".localized, avoidConflictsWith: [], choiceHandler: { [weak self] (selectedDirectory) in
+
+						if let targetDirectory = selectedDirectory {
+							self?.importFile(url: url, to: targetDirectory, core: core)
+						}
+					})
+
+					let pickerNavigationController = ThemeNavigationController(rootViewController: directoryPickerViewController)
+
+					if let wd = UIApplication.shared.delegate?.window {
+						let vc = wd!.rootViewController
+						if let navCon = vc as? UINavigationController, let topVC = navCon.visibleViewController {
+							topVC.present(pickerNavigationController, animated: true)
+						} else {
+							vc?.present(pickerNavigationController, animated: true)
+						}
+					}
+				}
+			}
+		})
+	}
+
+	func importFile(url : URL, to targetDirectory : OCItem, core : OCCore?) {
+		let name = url.lastPathComponent
+		if let progress = core?.importFileNamed(name,
+												at: targetDirectory,
+												from: url,
+												isSecurityScoped: false,
+												options: [OCCoreOption.importByCopying : true,
+														  OCCoreOption.automaticConflictResolutionNameStyle : OCCoreDuplicateNameStyle.bracketed.rawValue],
+												placeholderCompletionHandler: { (error, item) in
+													if error != nil {
+														Log.debug("Error uploading \(Log.mask(name)) to \(Log.mask(targetDirectory.path)), error: \(error?.localizedDescription ?? "" )")
+													}
+		},
+												resultHandler: { (error, _ core, _ item, _) in
+													if error != nil {
+														Log.debug("Error uploading \(Log.mask(name)) to \(Log.mask(targetDirectory.path)), error: \(error?.localizedDescription ?? "" )")
+													} else {
+														Log.debug("Success uploading \(Log.mask(name)) to \(Log.mask(targetDirectory.path))")
+
+														let fileManager = FileManager.default
+														do {
+															try fileManager.removeItem(at: url)
```

Comment:
If `true` was passed for [`UIApplicationOpenURLOptionsOpenInPlaceKey`](https://developer.apple.com/documentation/uikit/uiapplicationopenurloptionsopeninplacekey?language=objc), this would delete the original file. Please make sure to honor the value passed for [`UIApplicationOpenURLOptionsOpenInPlaceKey`](https://developer.apple.com/documentation/uikit/uiapplicationopenurloptionsopeninplacekey?language=objc) before deleting the file (possibly by using the approach I outlined in the review comment in `AppDelegate.swift`).